packages/expect/src/asymmetricMatchers.ts

```diff
@@ -146,14 +145,15 @@ class ArrayContaining extends AsymmetricMatcher {
 
 class ObjectContaining extends AsymmetricMatcher {
   sample: Object;
```

Comment:
type this as `{[key: string]: unknown}` (or maybe `any`), and you shouldn't have to do the type casting later